brave/brave-core
# [iOS] VPN Update UX with advanced server selection

Resolves https://github.com/brave/brave-browser/issues/36409

This PR is aimed at implementing making the server selection more detailed for VPN users. Especially with the increasing number servers added the idea is to divide the servers into sections of Country and City and also give the user the ability to choose which city they want to join in server list.

### The VPN Selection Changes

The change is beginning with removing the server list when automatic selection is turned on but showing the country used for automatic selection.

When Automatic selection is disabled the country selected using automatic selection will be manually selected.

Clicking a Country Name (anywhere except the information 'i' button) will choose the country and do automatic server selection for that country.

Clicking 'i' button will list the cities in that country and will give the option for optimal. Optimal is selecting the country working the same way but selecting any city should select that city.

![countrycity122 22](https://github.com/brave/brave-core/assets/6643505/8fa89f33-a737-4c80-b0f4-1bb144523b0a)

### Other Entry Point

Another entry point for the server selection is under VPN settings, this part is now showing flag, country name and city.

### Folder Structure

Additionally the folder structure is altered for BraveVPN and extension are added for the manager class to properly group the functionality into separate pieces. This also includes deleting the unused files and their usage inside the application.

New structure

![Screenshot 2024-06-18 at 5 36 21 PM](https://github.com/brave/brave-core/assets/6643505/e8d9a6e3-c466-4254-baf4-76f0b4588078)

### Localization

Localization is added and the Strings that are not used shared are relocated to String file inside the BraveVPN package.

Security Issue: https://github.com/brave/reviews/issues/1662



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Install Brave
- Buy VPN and install profile
- Check VPN is connected to Automatic Server
- Check server detail

https://github.com/brave/brave-core/assets/6643505/14c3e1fe-aa3d-4709-a0ce-7b1e2e2aac16

- Change Automatic server selection
- Change server to a new country
- Change server to a city
- Check server detail

https://github.com/brave/brave-core/assets/6643505/d27c2546-5da2-461d-975a-61613282bad6
